base/client.jl

```diff
@@ -323,6 +323,16 @@ function exec_options(opts)
             end
         end
     end
+
+    ret = 0
+    if isdefined(Main, :main)
```

Comment:
```suggestion
    if repl && isdefined(Main, :main)
```
?

Reply:
```suggestion
    if !isinteractive && isdefined(Main, :main)
```
suggestion from triage